airbyte-integrations/bases/base-python/base_python/sdk/streams/core.py

```diff
@@ -0,0 +1,104 @@
+import inspect
+from abc import ABC, abstractmethod
+from typing import Mapping, Any, Iterable, List, Union
+
+from airbyte_protocol import AirbyteStream, SyncMode
+from base_python.logger import AirbyteLogger
+from base_python.schema_helpers import ResourceSchemaLoader
+
+
+def package_name_from_class(cls: object) -> str:
+    """Find the package name given a class name"""
+    module = inspect.getmodule(cls)
+    return module.__name__.split(".")[0]
+
+class Stream(ABC):
+    # Use self.logger in subclasses to log any messages
+    logger = AirbyteLogger()  # TODO use native "logging" loggers with custom handlers
+
+    @property
+    def name(self) -> str:
+        """
+        :return: Stream name. By default this is the implementing class name, but it can be overridden as needed.
```

Comment:
should it be stream name transformed from camel case to snake case? most stream names are snake cased by convention, i think?

Reply:
```python
return inflection.underscore(name)
```

Reply:
added snake casing by default